### PR TITLE
[MRG] ENH/DOC: demo custom spectrum creation

### PR DESCRIPTION
**Pull request by dengemann**
#### What does this implement/fix?
Explain your changes.

A simple example of how to put custom power spectral data into an average Spectrum class.
A proper SpectrumArray class could follow if there is interest.
 
#### Additional information
Any additional information you think is important.

Discussed with @drammock  @hoechenberger in past 2 office hours.

### CODE REVIEW

File: tutorials/simulation/10_array_objs.py [hunk 1]
```diff
@@ -195,3 +195,48 @@
                                nave=data.shape[0], comment='simulated')
 print(evoked_array)
 evoked_array.plot()
+
+
+# %%
+# In certain situations you may wish to use a custom time-frequency
+# decomposition for estimation of power spectra. Or you may wish to
+# to process pre-computed power spectra in MNE.
```
Comment (hoechenberger) [inline tutorials/simulation/10_array_objs.py:203]: one "to" too many
```suggestion
# process pre-computed power spectra in MNE.
```

File: tutorials/simulation/10_array_objs.py [hunk 2]
```diff
@@ -195,3 +195,48 @@
...
+# Following the same logic, it is possible to instantiate averaged power
+# spectrum using the `~mne.time_frequency.Spectrum` class. This is slightly
```
Comment (hoechenberger) [inline tutorials/simulation/10_array_objs.py:205]:
```suggestion
# spectrum using the :class:`~mne.time_frequency.Spectrum` class.
# This is slightly
```

File: tutorials/simulation/10_array_objs.py [hunk 3]
```diff
@@ -195,3 +195,49 @@
...
+                        ) -> mne.time_frequency.Spectrum:  # Spectrum object
+    """Create MNE averaged power spectrum object from custom data"""
+    state = dict(
+        method='my_welch',
+        data=data,
+        sfreq=inst_info['sfreq'],
+        dims=('channel', 'freq'),
+        freqs=freqs,
+        inst_type_str='Raw',
+        data_type='simulated signals',
```
Comment (dengemann) [inline tutorials/simulation/10_array_objs.py:231]: @drammock I put this for illustration purposes. Are there any expectations in the Spectrum code regarding this field? It was 'Averaged EEG' in my concrete case on which this example is based.

Comment (drammock) [reply tutorials/simulation/10_array_objs.py:231]: right now it is only used in the `repr` of the object. I don't have any concrete plans to use it in any other way (and I would actively *avoid* using it in `if` clauses to triage different flavors of spectrum, it's probably not reliable to do so)

Comment (drammock) [reply tutorials/simulation/10_array_objs.py:231]: native values for this are `Fourier Coefficients`, `Power Spectrum`, and `Averaged Power Spectrum`. `Averaged Fourier Coefficients` is not supported, see 

https://github.com/mne-tools/mne-python/blob/18474b1d9227778ae875692a56f7e0927e856a54/mne/time_frequency/spectrum.py#L1109-L1118

**Comment by drammock** [comment]
CircleCI failure is this:
```
Traceback (most recent call last):
  File "/home/circleci/project/tutorials/simulation/10_array_objs.py", line 243, in <module>
    spectrum.plot(picks=[0, 1])
  File "/home/circleci/project/mne/time_frequency/spectrum.py", line 606, in plot
    _plot_psd(self, fig, self.freqs, psd_list, picks_list, titles_list,
  File "/home/circleci/project/mne/viz/utils.py", line 2297, in _plot_psd
    _plot_lines(psd_array, info, picks, fig, ax_list, spatial_colors,
  File "/home/circleci/project/mne/viz/evoked.py", line 472, in _plot_lines
    warn('Channel locations not available. Disabling spatial '
  File "/home/circleci/project/mne/utils/_logging.py", line 400, in warn
    warnings.warn_explicit(
RuntimeWarning: Channel locations not available. Disabling spatial colors.
```
it's just a warning, but it's annoying when this breaks our CIs. Makes me wonder if we should either (1) make it a `logger.info` or (2) make `spatial_colors='auto'` somehow.

**Review by drammock — APPROVED** [review]
changes LGTM. +1 for merge when CIs are green. 

My one hesitation is about `data_type`: I might be more comfortable sticking with `Power Spectrum` here, since our tutorials are implicitly supposed to demonstrate best practices and `Power Spectrum` would be the most consistent description of what the data are.

**Comment by dengemann** [comment]
Comments addressed, CIs green. Good to go from my side.

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
On,y thing I'd change is the indentation of the custom function header, I would blackify it

File: tutorials/simulation/10_array_objs.py [hunk 4]
```diff
@@ -195,3 +195,49 @@
...
+
+psd_ave = psd.mean(0)
+
+# map to `~mne.time_frequency.Spectrum` class and explore API
+
+
+def spectrum_from_array(data: np.ndarray,  # spectral features
+                        freqs: np.ndarray,  # frequencies
+                        inst_info: mne.Info  # the meta data of MNE instance
+                        ) -> mne.time_frequency.Spectrum:  # Spectrum object
```
Comment (hoechenberger) [inline tutorials/simulation/10_array_objs.py:222]:
```suggestion
def spectrum_from_array(
    data: np.ndarray,  # spectral features
    freqs: np.ndarray,  # frequencies
    inst_info: mne.Info  # the meta data of MNE instance
) -> mne.time_frequency.Spectrum:  # Spectrum object
```

**Review by hoechenberger — APPROVED** [review]

**Pull request merged**